packages/block-library/src/navigation/edit/index.js

```diff
@@ -256,16 +264,26 @@ function Navigation( {
 		setRef( fallbackNavigationMenus[ 0 ].id );
 	}, [ navigationMenus ] );
 
-	const navRef = useRef();
+	useEffect( async () => {
```

Comment:
As I mentioned during our pairing call on this feature, using `async` in an effect is not best practice. Why?

Because the effect is supposed to return a cleanup function, However `async` will always return a `Promise`. While this doesn't break in React17, it's not a good practice and it will start breaking in React 18.

An alternative is something like

```js
useEffect(() => {
    (async () => {
        // do stuff
    })();
  }, []);
```

Here is [a previous PR which had to clear up the same problem](https://github.com/WordPress/gutenberg/pull/40373).